alright, failure is the same on linux. It appears to be because nix is trying to install this package twice with different versions
```
4.drv'
litellm> Sourcing python-remove-tests-dir-hook
litellm> Sourcing python-catch-conflicts-hook.sh
litellm> Sourcing python-remove-bin-bytecode-hook.sh
litellm> Sourcing pypa-build-hook
litellm> Using pypaBuildPhase
litellm> Sourcing pypa-install-hook
litellm> Using pypaInstallPhase
litellm> Sourcing python-imports-check-hook.sh
litellm> Using pythonImportsCheckPhase
litellm> Sourcing python-namespaces-hook
litellm> Sourcing python-catch-conflicts-hook.sh
litellm> unpacking sources
litellm> unpacking source archive /nix/store/swjxpnnky2fcnjdk8nwzfvb6zni8v7kg-source
litellm> source root is source
litellm> setting SOURCE_DATE_EPOCH to timestamp 315619200 of file source/pyproject.toml
litellm> patching sources
litellm> configuring
litellm> no configure script, doing nothing
litellm> building
litellm> Executing pypaBuildPhase
litellm> Creating a wheel...
litellm> * Getting build dependencies for wheel...
litellm> * Building wheel...
litellm> Successfully built litellm-0.1.584-py3-none-any.whl
litellm> Finished creating a wheel...
litellm> Finished executing pypaBuildPhase
litellm> installing
litellm> Executing pypaInstallPhase
litellm> Successfully installed litellm-0.1.401-py3-none-any.whl
litellm> Traceback (most recent call last):
litellm>   File "/nix/store/m4kmky8ay96qwyakxjmj1idcvrdmvb9k-python3-3.10.12/lib/python3
.10/runpy.py", line 196, in _run_module_as_main
litellm>     return _run_code(code, main_globals, None,
litellm>   File "/nix/store/m4kmky8ay96qwyakxjmj1idcvrdmvb9k-python3-3.10.12/lib/python3
.10/runpy.py", line 86, in _run_code
litellm>     exec(code, run_globals)
litellm>   File "/nix/store/l0fgaf81k8kwvs5s6nlhbc1kipl2qdj0-python3.10-installer-0.7.0/
litellm>     _main(sys.argv[1:], "python -m installer")
litellm>   File "/nix/store/l0fgaf81k8kwvs5s6nlhbc1kipl2qdj0-python3.10-installer-0.7.0/lib/python3.10/site-packages/installer/__main__.py", line 94, in _main
litellm>     installer.install(source, destination, {})
litellm>   File "/nix/store/l0fgaf81k8kwvs5s6nlhbc1kipl2qdj0-python3.10-installer-0.7.0/lib/python3.10/site-packages/installer/_core.py", line 109, in install
litellm>     record = destination.write_file(
litellm>   File "/nix/store/l0fgaf81k8kwvs5s6nlhbc1kipl2qdj0-python3.10-installer-0.7.0/lib/python3.10/site-packages/installer/destinations.py", line 207, in write_file
litellm>     return self.write_to_fs(scheme, path_, stream, is_executable)
litellm>   File "/nix/store/l0fgaf81k8kwvs5s6nlhbc1kipl2qdj0-python3.10-installer-0.7.0/lib/python3.10/site-packages/installer/destinations.py", line 167, in write_to_fs
litellm>     raise FileExistsError(message)
litellm> FileExistsError: File already exists: /nix/store/gq7a86sg8n10y2miqd23vcb8bb7agg5i-python3.10-litellm-0.1.584/lib/python3.10/site-packages/litellm/.DS_Store
```
not sure why.

--- a/pkgs/development/python-modules/litellm/default.nix
+++ b/pkgs/development/python-modules/litellm/default.nix
@@ -1,0 +1,53 @@
+{ lib
+, buildPythonPackage
+, fetchFromGitHub
+, poetry-core
+, importlib-metadata
+, openai
+, python-dotenv
+, tiktoken
+}:
+let
+  version = "0.1.590";
+in
+buildPythonPackage rec {
+  pname = "litellm";
+  format = "pyproject";
+  inherit version;
+
+  src = fetchFromGitHub {
+    owner = "BerriAI";
+    repo = "litellm";
+    rev = "7cb96e86b4753008cbf8d116aca514750e98d360";
+    hash = "sha256-ITMcwGjelNfNGnfBmmdu0Xwph4u0mxiFSfHnysUxWCQ=";
+  };
+
+  postPatch = ''
+    rm -rf dist
+  '';
+
+  nativeBuildInputs = [
+    poetry-core
+  ];
+
+  propagatedBuildInputs = [
+    importlib-metadata
+    openai
+    python-dotenv
+    tiktoken
+  ];
+
+  # the import check phase fails trying to do a network request to openai
+  # pythonImportsCheck = [ "litellm" ];
+
+  # no tests
+  doCheck = false;
+
+  meta = with lib; {
+    description = "Use any LLM as a drop in replacement for gpt-3.5-turbo. Use Azure, OpenAI, Cohere, Anthropic, Ollama, VLLM, Sagemaker, HuggingFace, Replicate (100+ LLMs)";
+    homepage = "https://github.com/BerriAI/litellm";
+    license = licenses.mit;
+    changelog = "https://github.com/BerriAI/litellm/releases/tag/v${version}";
+    maintainers = with maintainers; [ happysalada ];
+  };
+}

--- a/pkgs/development/python-modules/tokentrim/default.nix
+++ b/pkgs/development/python-modules/tokentrim/default.nix
@@ -1,0 +1,39 @@
+{ lib
+, buildPythonPackage
+, fetchFromGitHub
+, poetry-core
+, tiktoken
+}:
+
+buildPythonPackage {
+  pname = "tokentrim";
+  version = "unstable-2023-09-07";
+  format = "pyproject";
+
+  src = fetchFromGitHub {
+    owner = "KillianLucas";
+    repo = "tokentrim";
+    rev = "e98ad3a2ca0e321a7347f76c30be584175495139";
+    hash = "sha256-95xitHnbFFaj0xPuLMWvIvuJzoCO3VSd592X1RI9h3A=";
+  };
+
+  nativeBuildInputs = [
+    poetry-core
+  ];
+
+  propagatedBuildInputs = [
+    tiktoken
+  ];
+
+  pythonImportsCheck = [ "tokentrim" ];
+
+  # tests connect to openai
+  doCheck = false;
+
+  meta = with lib; {
+    description = "Easily trim 'messages' arrays for use with GPTs";
+    homepage = "https://github.com/KillianLucas/tokentrim";
+    license = licenses.mit;
+    maintainers = with maintainers; [ happysalada ];
+  };
+}

--- a/pkgs/tools/llm/open-interpreter/default.nix
+++ b/pkgs/tools/llm/open-interpreter/default.nix
@@ -1,0 +1,52 @@
+{ lib
+, python3
+, fetchFromGitHub
+}:
+let
+  version = "0.1.3";
+in
+python3.pkgs.buildPythonApplication {
+  pname = "open-interpreter";
+  format = "pyproject";
+  inherit version;
+
+  src = fetchFromGitHub {
+    owner = "KillianLucas";
+    repo = "open-interpreter";
+    rev = "v${version}";
+    hash = "sha256-xmmyDIshEYql41k/7gF+ay7s3mI+iGCjr5gDfLkqLU0=";
+  };
+
+  nativeBuildInputs = [
+    python3.pkgs.poetry-core
+  ];
+
+  propagatedBuildInputs = with python3.pkgs; [
+    appdirs
+    astor
+    gitpython
+    huggingface-hub
+    inquirer
+    litellm
+    openai
+    # pyreadline3 # this is a windows deps
+    python-dotenv
+    rich
+    six
+    tiktoken
+    tokentrim
+    wget
+  ];
+
+  # the import check phase fails trying to do a network request to openai
+  # because of litellm
+  # pythonImportsCheck = [ "interpreter" ];
+
+  meta = with lib; {
+    description = "OpenAI's Code Interpreter in your terminal, running locally";
+    homepage = "https://github.com/KillianLucas/open-interpreter";
+    license = licenses.mit;
+    changelog = "https://github.com/KillianLucas/open-interpreter/releases/tag/v${version}";
+    maintainers = with maintainers; [ happysalada ];
+  };
+}

--- a/pkgs/top-level/all-packages.nix
+++ b/pkgs/top-level/all-packages.nix
@@ -11541,6 +11541,8 @@ with pkgs;
 
   open-ecard = callPackage ../tools/security/open-ecard { };
 
+  open-interpreter = callPackage ../tools/llm/open-interpreter { };
+
   openjade = callPackage ../tools/text/sgml/openjade { };
 
   openhantek6022 = libsForQt5.callPackage ../applications/science/electronics/openhantek6022 { };

--- a/pkgs/top-level/python-packages.nix
+++ b/pkgs/top-level/python-packages.nix
@@ -12933,6 +12933,8 @@ self: super: with self; {
 
   tokenlib = callPackage ../development/python-modules/tokenlib { };
 
+  tokentrim = callPackage ../development/python-modules/tokentrim { };
+
   tololib = callPackage ../development/python-modules/tololib { };
 
   toml = callPackage ../development/python-modules/toml { };

--- a/pkgs/top-level/python-packages.nix
+++ b/pkgs/top-level/python-packages.nix
@@ -6188,6 +6188,8 @@ self: super: with self; {
 
   lit = callPackage ../development/python-modules/lit { };
 
+  litellm = callPackage ../development/python-modules/litellm { };
+
   litemapy = callPackage ../development/python-modules/litemapy { };
 
   littleutils = callPackage ../development/python-modules/littleutils { };

